packages/bee-q/src/components/breadcrumb-item/bq-breadcrumb-item.tsx

```diff
@@ -0,0 +1,153 @@
+  /** Handler to be called when item loses focus */
+  @Event() bqBlur: EventEmitter<HTMLBqBreadcrumbItemElement>;
+
+  /** Handler to be called when item is focused */
+  @Event() bqFocus: EventEmitter<HTMLBqBreadcrumbItemElement>;
+
+  /** Handler to be called when item is clicked */
+  @Event() bqClick: EventEmitter<HTMLBqBreadcrumbItemElement>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private onBlur = () => {
+    this.bqBlur.emit(this.el);
+  };
+
+  private onFocus = () => {
+    this.bqFocus.emit(this.el);
+  };
+
+  private onClick = () => {
+    this.bqClick.emit(this.el);
+  };
+
+  private onSlotChange = () => {
+    this.hasPrefix = hasSlotContent(this.prefixElem, 'prefix');
+    this.hasSuffix = hasSlotContent(this.suffixElem, 'suffix');
+    this.hasLabel = hasSlotContent(this.labelElem, 'label');
+  };
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    const isLink = isDefined(this.href);
+    const TagElem = isLink ? 'a' : 'button';
+    const separatorElem = this.separatorIcon ? <bq-icon size="12" name={this.separatorIcon}></bq-icon> : '/';
+
+    return (
+      <section class="flex h-5 items-center" role="listitem">
+        <TagElem
+          class="breadcrumb-item"
+          tabindex="0"
+          href={isLink ? this.href : undefined}
+          rel={isLink && this.target ? 'noreferrer noopener' : undefined}
+          target={isLink ? this.target : undefined}
+          onBlur={this.onBlur}
+          onFocus={this.onFocus}
+          onClick={this.onClick}
+        >
+          <span class="breadcrumb-item__prefix" ref={(elem) => (this.prefixElem = elem)} part="prefix">
+            <slot name="prefix" onSlotchange={this.onSlotChange}></slot>
+          </span>
+          <span
+            class={{
+              'breadcrumb-item__label': true,
+              'text-text-brand': this.hideSeparatorIcon,
+              'pl-xs': this.hasPrefix && this.hasLabel,
+              'pr-xs': this.hasSuffix && this.hasLabel,
+            }}
+            ref={(elem) => (this.labelElem = elem)}
+            part="label"
+          >
+            <slot name="label" onSlotchange={this.onSlotChange}></slot>
+          </span>
+          <span class="breadcrumb-item__suffix" ref={(elem) => (this.suffixElem = elem)} part="suffix">
+            <slot name="suffix" onSlotchange={this.onSlotChange}></slot>
+          </span>
```

Comment:
I wonder if we really need 3 slots here 🤔 ... I think it will be easier to have one slot and allows consumers to create the breadcrumb layout as they wish:
- just label
- prefix and label
- label and suffix
- prefix, label, and suffix

Something like:

```html
<bq-breadcrumb-item>
  <div class="flex gap-m">
    <bq-icon name="house"></bq-icon>
    <span>Home</span>
  </div>
</bq-breadcrumb-item>
```

making consumers responsible for applying the CSS rules over the content inside the breadcrumb?

Any thoughts on this @endv-bogdanb ?

Reply:
I also prefer to keep things simple whenever it's possible. Here, I followed the design from Figma, where we can have items displaying only 1 of the 3 parts, so only icon or only label or all of them, but for this particular component, I think that we can simplify things and allow the consumer to add their preferred spacing.